api/src/main/java/edu/wpi/first/shuffleboard/api/util/NetworkTableUtils.java

```diff
@@ -119,6 +119,14 @@ public static boolean isDelete(int flags) {
   }
 
   /**
+   * Checks if the given key is metadata, eg matches the format "~METADATA~" or ".metadata"
+   */
+  public static boolean isMetadata(String key) {
+    return key.matches("^.*/\\..+(/.*)?$")
```

Comment:
I should precompile this as a `Pattern`, actually. But yeah, the regex could be cleaner